docs/source/reference-io.rst

```diff
@@ -631,6 +631,11 @@ Asynchronous path objects
 
 .. autoclass:: Path
    :members:
+   :inherited-members:
+
+.. autoclass:: PosixPath
+
+.. autoclass:: WindowsPath
```

Comment:
What is the use case of using `trio.WindowsPath` or `trio.PosixPath`? Looking at the code, it looks like we only wrap methods from `PurePath` so it's not like they get more functionality (?)

I'm probably missing something trivial though. Though if it's not necessary to then I think having to put platform-specific types is weird for an exported API.

Reply:
If you look at the `pathlib` module, that's exactly what they do there. Windows and posix have very different ways of expressing a path (backslashes vs forwardslashes etc). This is making the `trio` path behavior just like the stdlib.

Practically, being able to make `trio.Path` a subclass of `PurePath` means it interops with all other path-derived things, and dramatically simplifies the implementation. However, you can't directly subclass `PurePath`... you have to subclass `PureWindowsPath` or `PurePosixPath`. And that means you need to have to concrete `PosixPath` and `WindowsPath` classes.